### PR TITLE
Account for ProxyTracerProvider

### LINKED ISSUE #405: [redis] Unable to initiate tracer
**Describe your environment** 
Unable to initiate tracer

**Steps to reproduce**
```
 ## Redis
 from opentelemetry.instrumentation.redis import RedisInstrumentor
 import redis
 RedisInstrumentor().instrument()
 redis_client = redis.StrictRedis(host="localhost", port=6379)
 ```

**What is the expected behavior?**
tracer instance

**What is the actual behavior?**
```
Overriding of current TracerProvider is not allowed
Traceback (most recent call last):
  File "/tmp/tracing/example/scrubber_app.py", line 9, in <module>
    tracer = utils.init_tracer('sample_app')
  File "/tmp/utils.py", line 40, in init_tracer
    trace.get_tracer_provider().add_span_processor(
AttributeError: '_DefaultTracerProvider' object has no attribute 'add_span_processor'
```

**Additional context**
Reference: https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

### PR DESCRIPTION
**Pull request by owais**
# Description

Updated tests that relied on unset tracer being a noop tracer instead of a proxy tracer.

Includes changes from #400 to make the tests pass.

Fixes #405 

## Type of change

Please delete options that are not relevant.

- [x] Fixed broken tests due to change in core repo
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Fixed tests

# Does This PR Require a Core Repo Change?

- [x] Yes. - Link to PR: https://github.com/open-telemetry/opentelemetry-python/pull/1726
- [ ] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-requests/tests/test_requests_integration.py [hunk 1]
```diff
@@ -177,10 +177,8 @@ def test_suppress_instrumentation(self):
     def test_not_recording(self):
         with mock.patch("opentelemetry.trace.INVALID_SPAN") as mock_span:
             RequestsInstrumentor().uninstrument()
-            # original_tracer_provider returns a default tracer provider, which
-            # in turn will return an INVALID_SPAN, which is always not recording
             RequestsInstrumentor().instrument(
-                tracer_provider=self.original_tracer_provider
+                tracer_provider=trace._DefaultTracerProvider
```
Comment (owais) [inline instrumentation/opentelemetry-instrumentation-requests/tests/test_requests_integration.py:181]: Looks like the intention was to ensure that the instrumentation works with noop tracers.

Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-requests/tests/test_requests_integration.py:181]: Where is `trace.__DefaultTracerProvider`?

Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-requests/tests/test_requests_integration.py:181]: Also should probably remove `original_tracer_provider` from [TestBase](https://github.com/open-telemetry/opentelemetry-python/blob/main/tests/util/src/opentelemetry/test/test_base.py#L29)

Comment (owais) [reply instrumentation/opentelemetry-instrumentation-requests/tests/test_requests_integration.py:181]: It is not directly used but I think might still be useful as it remembers the registered provider before a test case runs and automatically restores it back after a test case finishes running.

**Review by lzchen — APPROVED** [review]

**Pull request merged**